samples/javascript_typescript/60.multilingual-luis-bot/translator.ts

```diff
@@ -0,0 +1,345 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as request from 'request-promise-native';
+import { promises } from 'fs';
+
+
+export class TranslateArrayOptions{
+    texts: string[];    from: string;
+    to: string;
+    contentType?: string;
+    category?: string;
+}
+
+/**
+ * @private
+ */
+export class TranslationResult {
+    translatedText: string;
+}
+
+/**
+ * @private
+ */
+export class TranslationRequest{
+	Text: string;
+}
+
+/**
+ * @private
+ */
+export class MicrosoftTranslator {
+    
+    apiKey: string;
+    postProcessor: TranslatorPostProcessor;
+
+    constructor(apiKey: string, noTranslatePatterns?: {[id: string] : string[]}, wordDictionary?: { [id: string]: string }) {
+        this.apiKey = apiKey;
+        this.postProcessor = new TranslatorPostProcessor();
+        this.postProcessor = new TranslatorPostProcessor(noTranslatePatterns, wordDictionary);
+
+    }
+
+    public detect = async (text: string): Promise<object> => {
+        let uri: any = "https://api.cognitive.microsofttranslator.com/detect?api-version=3.0";
+        let translationRequest = new TranslationRequest();
+        translationRequest.Text = text;
+        const response = await request({
+            url: uri,
+            method: 'POST',
+            headers: {
+                'Ocp-Apim-Subscription-Key': this.apiKey,
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify([translationRequest]),
+        });
+        let detectedLanguage = JSON.parse(response);
+        return detectedLanguage;
+    };
+
+    public translateArray = async (options: TranslateArrayOptions):  Promise<object> => {
+		try {
+            let from = options.from;
+            let to = options.to;
+            let texts = options.texts;
+            let orgTexts = [];
+            let translationRequests = [];
+            let results = [];
+            texts.forEach((text, index, array) => {
+                let normalizedText = text.trim();
+                    orgTexts.push(normalizedText);
+                    texts[index] = normalizedText;
+                    let translationRequest = new TranslationRequest();
+                    translationRequest.Text = normalizedText;
+                    translationRequests.push(translationRequest);
+            });
+            let uri: any = `https://api.cognitive.microsofttranslator.com/translate?api-version=3.0&includeAlignment=true&includeSentenceLength=true&from=${from}&to=${to}`;
+			const response = await request({
+				url: uri,
+				method: 'POST',
+				headers: {
+                    'Ocp-Apim-Subscription-Key': this.apiKey,
+					'Content-Type': 'application/json',
+				},
+				body: JSON.stringify(translationRequests),
+            });
+            let translations = JSON.parse(response);
+            translations.forEach((traslationModel, index, array) => {
```

Comment:
```suggestion
            translations.forEach((traslationModel, index) => {
```
Remove unused variable